clients/src/main/java/org/apache/kafka/common/requests/ConsumerGroupDescribeRequest.java

```diff
@@ -63,12 +63,14 @@ public ConsumerGroupDescribeRequest(ConsumerGroupDescribeRequestData data, short
     public ConsumerGroupDescribeResponse getErrorResponse(int throttleTimeMs, Throwable e) {
         ConsumerGroupDescribeResponseData data = new ConsumerGroupDescribeResponseData()
             .setThrottleTimeMs(throttleTimeMs);
+        ApiError apiError = ApiError.fromThrowable(e);
         // Set error for each group
         this.data.groupIds().forEach(
             groupId -> data.groups().add(
                 new ConsumerGroupDescribeResponseData.DescribedGroup()
                     .setGroupId(groupId)
-                    .setErrorCode(Errors.forException(e).code())
+                    .setErrorCode(apiError.code())
+                    .setErrorMessage(apiError.messageWithFallback())
```

Comment:
Using `messageWithFallback` here seems to beat the intention of wrapping the exception in an `ApiError` I would say. `ApiError` carefully determines the message to show (null if it considers it shouldn't), so I wonder if we should just align and `setErrorMessage(apiError.message())` only `if apiError.message() != null` .

Ex. `ApiError` intentionally returns a null message for UNKNOWN_SERVER_ERROR, to avoid leaking sensitive info. Us falling back to the error message here means that we would still include the error message (potentially sensitive) and include it in the `ConsumerGroupDescribeResponse`. Makes sense?

Reply:
Sorry for the late response. I'm not sure it works if we use `apiError.message`. The original idea of this pr is trying to give end-user a friendly error msg, i.e., the default error msg defined at Errors rather than null... So I changed to use `apiError.messageWithFallback`.

If we change to use `apiError.message`, then nothing changes since it always returns null for default error msg.